src/test/java/seedu/address/logic/commands/RemoveCommandTest.java

```diff
@@ -0,0 +1,146 @@
+
+        RemovePersonDescriptor descriptor = new RemovePersonDescriptorBuilder().withLanguages("Python").build();
+        RemoveCommand removeCommand = new RemoveCommand(indexLastPerson, descriptor);
+
+        String expectedMessage = String.format(RemoveCommand.MESSAGE_REMOVE_FIELD_SUCCESS, RemovedFieldPerson);
+
+        Model expectedModel = new ModelManager(new AddressBook(model.getAddressBook()), new UserPrefs());
+        expectedModel.setPerson(lastPerson, RemovedFieldPerson);
+
+        assertCommandSuccess(removeCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_clearTagFieldSpecifiedUnfilteredList_success() {
+        Index indexBenson = Index.fromOneBased(2);
+        Person Benson = model.getFilteredPersonList().get(indexBenson.getZeroBased());
+
+        PersonBuilder personInList = new PersonBuilder();
+        Person RemovedFieldPerson = personInList.withName("Benson Meier")
+                .withProfile("benson-meier")
+                .withAddress("311, Clementi Ave 2, #02-25")
+                .withEmail("johnd@example.com").withPhone("98765432")
+                .withLanguages("Java", "JavaScript").build();
+
+        RemovePersonDescriptor descriptor = new RemovePersonDescriptorBuilder().withTags().build();
+        RemoveCommand removeCommand = new RemoveCommand(indexBenson, descriptor);
+
+        String expectedMessage = String.format(RemoveCommand.MESSAGE_REMOVE_FIELD_SUCCESS, RemovedFieldPerson);
+
+        Model expectedModel = new ModelManager(new AddressBook(model.getAddressBook()), new UserPrefs());
+        expectedModel.setPerson(Benson, RemovedFieldPerson);
+
+        assertCommandSuccess(removeCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_invalidPersonIndexUnfilteredList_failure() {
+        Index outOfBoundIndex = Index.fromOneBased(model.getFilteredPersonList().size() + 1);
+        RemovePersonDescriptor descriptor = new RemovePersonDescriptorBuilder().build();
+        RemoveCommand editCommand = new RemoveCommand(outOfBoundIndex, descriptor);
+
+        assertCommandFailure(editCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
+
+    /**
+     * Edit filtered list where index is larger than size of filtered list,
+     * but smaller than size of address book
+     */
+    @Test
+    public void execute_invalidPersonIndexFilteredList_failure() {
+        showPersonAtIndex(model, INDEX_FIRST_PERSON);
+        Index outOfBoundIndex = INDEX_SECOND_PERSON;
+        // ensures that outOfBoundIndex is still in bounds of address book list
+        assertTrue(outOfBoundIndex.getZeroBased() < model.getAddressBook().getPersonList().size());
+
+        RemoveCommand editCommand = new RemoveCommand(outOfBoundIndex,
+                new RemovePersonDescriptorBuilder().build());
+
+        assertCommandFailure(editCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+    }
+
+
+}
```

Comment:
Yes, current implementation should accept flags with inputs as the example given by you. The reason I do that is because I believe user might input the wrong flag with could essentially remove wanted field value of address/email/profile/phone. By doing so, if an user wants remove field value of address/email/profile/phone, he/she could `remove 1 a/` or `remove 1 a/someaddress` but if the `someaddress` is not exists, it wont get removed.

Reply:
Yes, _both_ `remove 1 a/` and `remove 1 a/someaddress` would remove `someaddress` under the address field.

For my understanding, the removeCommandParser only helps to store the field indicate by the user, so removeCommandParser should do similar job to the original version of editCommandParser.

Reply:
Noted, I was just curious where the testing for the parsing of the commands would be implemented, thanks!